geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java

```diff
@@ -63,6 +65,31 @@ private static ClientProtocolMessageHandler findClientProtocolMessageHandler() {
     }
   }
 
+  private static Class<? extends StreamAuthenticator> findStreamAuthenticator(
+      String implementationID) {
+    if (authenticatorClass != null) {
+      return authenticatorClass;
+    }
+
+    synchronized (streamAuthenticatorLoadLock) {
```

Comment:
Once `authenticatorClass` is initialized, there will no longer be any synchronization necessary in this method. This means that each thread will need to synchronize at most once, which means no locking / cache misses.